docker/local-development.md

```diff
@@ -0,0 +1,279 @@
+4. [A `tests/docker-compose.yml` file](#4-testsdocker-composeyml)
+that overrides the application service in the root file, in order to run the tests
+4. [A `docker-compose.db-ops.yml` file](#5-docker-composedb-opsyml-optional)
+that automates a multi-step data loading routine (Optional)
+
+With this setup, you can:
+
+- [Run your application or tests](#run-the-application)
+- [Run custom commands on your container](#run-custom-commands-on-containers)
+
+## Components
+
+### 1. `Dockerfile`
+
+```Dockerfile
+# Extend the base Python image
+# See https://hub.docker.com/_/python for version options
+# N.b., there are many options for Python images. We used the plain
+# version number in the pilot. YMMV. See this post for a discussion of
+# some options and their pros and cons:
+# https://pythonspeed.com/articles/base-image-python-docker-images/
+FROM python:<PYTHON_VERSION>
+
+# Give ourselves some credit
+LABEL maintainer "DataMade <info@datamade.us>"
+
+# Install and upgrade pip
+# This may not be necessary, depending on your base Python image
+RUN apt-get update
+RUN apt-get install -y python-pip
+RUN pip install --upgrade pip setuptools
+
+# Inside the container, create an app directory and switch into it
+RUN mkdir /app
+WORKDIR /app
+
+# Copy the requirements file into the app directory, and install them. Copy
+# only the requirements file, so Docker can cache this build step. Otherwise,
+# the requirements must be reinstalled every time you build the image after
+# the app code changes. See this post for further discussion of strategies
+# for building lean and efficient containers:
+# https://blog.realkinetic.com/building-minimal-docker-containers-for-python-applications-37d0272c52f3
+COPY ./requirements.txt /app/requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the contents of the current host directory (i.e., our app code) into
+# the container.
+COPY . /app
+```
+
+### 2. `scripts/init-db.sh`
+
+The Postgres image provides a harness for executing arbitrary SQL and Bash
+scripts when containers are initialized. [Read more. &raquo;](https://docs.docker.com/samples/library/postgres/#initialization-scripts)
+
+```bash
+#!/bin/bash
+set -e
+
+psql -U postgres -c "CREATE DATABASE <YOUR_DATABASE>"
+psql -U postgres -d <YOUR_DATABASE> -c "CREATE EXTENSION IF NOT EXISTS <YOUR_EXTENSION>"
+# ad inf.
+```
+
+### 3. `docker-compose.yml`
+
+```yaml
+version: '2.4'
+
+services:
+  app:
+    image: <YOUR_APP>
+    container_name: <YOUR_APP>
+    restart: always
+    build: .
+    ports:
+      # Map ports on your computer to ports on your container. This allows you,
+      # e.g., to visit your containerized application in a browser on your
+      # computer.
+      - <HOST_PORT>:<CONTAINER_PORT>  # e.g., 8000:8000
+    depends_on:
+      # Declare any services that should be started first. Beware: It checks
+      # that a service has started, but not that a service is ready.
```

Comment:
Now that we're using `condition: service_healthy`, I think this warning no longer applies, correct?